### PR TITLE
task07

### PR DESCRIPTION
**Pull request by JorgeMIng**

### CODE REVIEW

**Comment by anaigmo** [comment]
You need to merge in one PR this one and #35

File: Curso2023-2024/task07.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+# %% [markdown]
+# **TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL**
+
+# %%
+# TO DO
+# Visualize the results
+
+for s,p,o in g.triples((None,RDFS.subClassOf,ns.LivingThing)):
+    print(s)
```
Comment (dgarijo) [inline Curso2023-2024/task07.py:31]: You are not listing the subclasses of subclasses of living thing

File: Curso2023-2024/task07.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+
+# %%
+from rdflib.plugins.sparql import prepareQuery
+
+
+q1 = prepareQuery('''
+  SELECT ?subclass WHERE { 
+    ?subclass rdfs:subClassOf ns:LivingThing.
```
Comment (dgarijo) [inline Curso2023-2024/task07.py:40]: same as above. Use the `*` operator

File: Curso2023-2024/task07.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+       
+
+# %%
+q1 = prepareQuery('''
+  SELECT ?individual  WHERE { 
+    
+    {
+      ?individual rdf:type ?subclass.
+      optional{?subclass rdfs:subClassOf ?subclass.}
+      ?subclass rdfs:subClassOf ns:Person
```
Comment (dgarijo) [inline Curso2023-2024/task07.py:87]: Almost correct. Use rdf:type/rdfs:subclassOf* to make it fully recursive

File: Curso2023-2024/task07.py [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+# Visualize the results
+for person,p,o in g.triples((None,FOAF.knows,None)):
+  list=[]
+  for elem in g.triples((person,FOAF.knows,None)):
+    list.append(elem[2])
+  if len(list)>=2 :
+    for elem in list:
+      print(person,"knows",elem)
+  print("")
+
```
Comment (dgarijo) [inline Curso2023-2024/task07.py:171]: missing 7.5 in sparql

**Review by dgarijo — CHANGES_REQUESTED** [review]
PLease address remaining comments

**Comment by anaigmo** [comment]
Entregado con algún error, no se admiten más PR

**Pull request merged**